c/src/lib.rs

```diff
@@ -828,6 +858,24 @@ pub struct FoxgloveMcapOptions {
     pub emit_metadata_indexes: bool,
     pub repeat_channels: bool,
     pub repeat_schemas: bool,
+    /// Context provided to the `sink_channel_filter` callback.
+    pub sink_channel_filter_context: *const c_void,
+    /// A filter for channels that can be used to subscribe to or unsubscribe from channels.
+    ///
+    /// This can be used to omit one or more channels from a sink, but still log all channels to another
+    /// sink in the same context. Return false to disable logging of this channel.
+    ///
+    /// This method is invoked from the client's main poll loop and must not block.
+    ///
+    /// # Safety
+    /// - If provided, the handler callback must be a pointer to the filter callback function,
+    ///   and must remain valid until the server is stopped.
```

Comment:
Nit: is "server" the term we typically use with mcap writers?

Reply:
Nope, just a lazy copy/paste from above.